src/test/java/chambre/morsecode/WordleControllerTest.java

```diff
@@ -2,28 +2,110 @@
 
 import org.junit.jupiter.api.Test;
 
+import javax.swing.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
 class WordleControllerTest {
     @Test
     public void addLetter() {
-        //asserts the letter has been added to the word
+        //given
+        WordleGame game = mock();
+        WordleDictionary dictionary = mock();
+        JLabel[][] labels = new JLabel[][]{{
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        },
+        };
+        JButton enter = mock();
+        JButton back = mock();
+        WordleController controller = new WordleController(game, dictionary, labels, enter, back);
+
+        //when
+        controller.addLetter("A");
 
-        //asserts the letter shows up in the right grid box on the screen
+        //then
+        verify(labels[0][0]).setText("A");
+        assertEquals("A", controller.theGuess.toString());
     }
 
     @Test
     public void enterGuess() {
+        //given
+        WordleGame game = mock();
+        WordleDictionary dictionary = mock();
+        JLabel[][] labels = new JLabel[][]{{
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        },
+        };
+        JButton enter = mock();
+        JButton back = mock();
+        WordleController controller = new WordleController(game, dictionary, labels, enter, back);
+
+        //when
+        for (int i = 0; i < game.getWordleWord().length() - 1; i++) {
+            controller.addLetter(game.getWordleWord().substring(i, i + 1));
+        }
+        controller.enterGuess();
+
+        //then
         //asserts that all colors have changed to the correct color
+        verify(labels[0][0].getBackground());
```

Comment:
This is not the correct way to do a verify

Reply:
Yes, I realized I had a syntax error and fixed it.